File: library-core/src/main/java/com/mikepenz/fastadapter/utils/DefaultItemList.java
package com.mikepenz.fastadapter.utils;

import com.mikepenz.fastadapter.FastAdapter;
import com.mikepenz.fastadapter.IItem;
import com.mikepenz.fastadapter.IItemList;

/**
 * The default item list
 */

public abstract class DefaultItemList<Item extends IItem> implements IItemList<Item> {

    private FastAdapter<Item> fastAdapter;

    public void setFastAdapter(FastAdapter<Item> fastAdapter) {
        this.fastAdapter = fastAdapter;
    }

    public FastAdapter<Item> getFastAdapter() {
        return fastAdapter;
    }
}

File: library-core/src/main/java/com/mikepenz/fastadapter/utils/DefaultItemList.kt
package com.mikepenz.fastadapter.utils

import com.mikepenz.fastadapter.GenericItem
import com.mikepenz.fastadapter.IFastAdapter
import com.mikepenz.fastadapter.IItemList

/**
 * The default item list
 */

abstract class DefaultItemList<Item : GenericItem> : IItemList<Item> {

    var fastAdapter: IFastAdapter<Item>? = null
}
